graspy/inference/latent_distribution_test.py

```diff
@@ -14,121 +14,144 @@
+                msg = "n_components must be an int, not {}.".format(type(n_components))
                 raise TypeError(msg)
 
         if not isinstance(n_bootstraps, int):
-            msg = "n_bootstraps must an int, not {}".format(type(n_bootstraps))
+            msg = "n_bootstraps must be an int, not {}".format(type(n_bootstraps))
             raise TypeError(msg)
         elif n_bootstraps < 1:
             msg = "{} is invalid number of bootstraps, must be greater than 1"
             raise ValueError(msg.format(n_bootstraps))
 
-        if bandwidth is not None and not isinstance(bandwidth, float):
-            msg = "bandwidth must an int, not {}".format(type(bandwidth))
+        if not isinstance(num_workers, int):
+            msg = "num_workers must be an int, not {}".format(type(num_workers))
             raise TypeError(msg)
+        elif num_workers <= 0:
+            msg = "{} is invalid number of workers, must be greater than 0"
+            raise ValueError(msg.format(num_workers))
+        elif num_workers > 1:
+            raise NotImplementedError()  # TODO env error parallelizing
 
         super().__init__(embedding="ase", n_components=n_components)
+        self.distance = distance
+        self.test = KSample(test, compute_distance=distance)
         self.n_bootstraps = n_bootstraps
-        self.bandwidth = bandwidth
-
-    def _gaussian_covariance(self, X, Y):
-        diffs = np.expand_dims(X, 1) - np.expand_dims(Y, 0)
-        if self.bandwidth is None:
-            self.bandwidth = 0.5
-        return np.exp(-0.5 * np.sum(diffs ** 2, axis=2) / self.bandwidth ** 2)
-
-    def _statistic(self, X, Y):
-        N, _ = X.shape
-        M, _ = Y.shape
-        x_stat = np.sum(self._gaussian_covariance(X, X) - np.eye(N)) / (N * (N - 1))
-        y_stat = np.sum(self._gaussian_covariance(Y, Y) - np.eye(M)) / (M * (M - 1))
-        xy_stat = np.sum(self._gaussian_covariance(X, Y)) / (N * M)
-        return x_stat - 2 * xy_stat + y_stat
+        self.num_workers = num_workers
 
     def _embed(self, A1, A2):
+        if not is_symmetric(A1) or not is_symmetric(A2):
+            raise NotImplementedError()  # TODO asymmetric case
+
+        if self.n_components is None:
+            num_dims1 = select_dimension(A1)[0][-1]
+            num_dims2 = select_dimension(A2)[0][-1]
+            self.n_components = max(num_dims1, num_dims2)
+
         ase = AdjacencySpectralEmbed(n_components=self.n_components)
         X1_hat = ase.fit_transform(A1)
         X2_hat = ase.fit_transform(A2)
+
         if isinstance(X1_hat, tuple) and isinstance(X2_hat, tuple):
             X1_hat = np.concatenate(X1_hat, axis=-1)
             X2_hat = np.concatenate(X2_hat, axis=-1)
         elif isinstance(X1_hat, tuple) ^ isinstance(X2_hat, tuple):
             raise ValueError("Input graphs do not have same directedness")
+
         return X1_hat, X2_hat
 
-    def _median_heuristic(self, X1, X2):
+    def _median_sign_flips(self, X1, X2):
```

Comment:
can you explain this part

Reply:
i think the original function name isn't specific enough. I renamed it to describe that the function is flipping data (* -1) if the medians have different signs

Reply:
i meant can you explain what this is doing, not the change

Reply:
Minh mentioned that the misalignment of embeddings will be approximately off by sign flips, and that a cheap hack to verify this is to make sure the medians of each dimension have the same sign